### ISSUE TITLE
Consider using npm module "ignore"

### ISSUE THREAD

**Issue by cactysman** [patrick-steele-idem/ignoring-watcher#5]
Hello,

as good as the package `anymatch` may be, it itself - or the way you use it in this package - does not allow for exceptional path definitions as found in `.gitignore` files.

**Example**
```
foo/*
!foo/bar
```

Instead it adds the `!` as part of the path to be ignored.

Please consider using [`ignore`](https://www.npmjs.com/package/ignore) in addition to or instead of `anymatch`.
It allows using the default `.gitignore` syntax to ignore and un-ignore files. Maybe you could even turn its output into a custom `anymatch` matcher function so that everyone would be happy.

**Comment by jedwards1211** [comment]
agreed, I'm tempted to code up my own package that uses `chokidar` and `ignore`...will post back if I do
